src/helpers.ts

```diff
@@ -1755,8 +1760,20 @@ Note that in this case, the contract deployment will not behave the same if depl
                 throw error;
               }
             }
+
+            // make sure to close an existing connection before every transaction since it's currently not being handled
+            // properly by ethers
+            if (ledgerSigner) {
+              const __eth = await ledgerSigner._eth;
+              await __eth.transport.device.close();
```

Comment:
is it the right place to do it ?
It will only close on fetching `from`, what if 2 tx need to be executed using the same `from`

Reply:
Yeah, we have been using this to execute many transactions with the same `from` (in the format of`ledger://XXX`) in the same deployment script. It's ok to reuse the transport instance (which would happen in the case of the same `from` in the same deployment script). The original issue was that a new deployment script will try to connect/create a new transport instance without closing it first.